### DISCUSSION TITLE
Chiseled Docker image with ASP.NET Core and dotnet-monitor

### DISCUSSION THREAD

**Discussion by mu88** [dotnet/dotnet-monitor#6957] [Q&A] [answered]
Hi 👋🏻

I'm running several ASP.NET Core apps based on the chiseled Docker image. I'd like to integrate `dotnet-monitor` into my apps so that I can easily create memory dumps.

What's the best way to accomplish that? From my current understanding, I'd have to build my own base image, combining both chiseled ASP.NET Core and `dotnet-monitor`.

Is that correct or is there an easier way, e.g. a pre-built Docker image from you guys that I can just grab? 

Thx!

**Comment by jander-msft** [comment, marked as answer]
It depends on what your orchestration environment is; for example, if you are using Kubernetes or Docker compose (something that allows for multiple containers and allows for volume mounts), then you can run .NET Monitor as a sidecar without altering your application Docker image. See https://github.com/dotnet/dotnet-monitor/blob/main/documentation/kubernetes.md for an example of how to do this Kubernetes.
